include/ensmallen_bits/pso/init_policies/constrained_init.hpp

```diff
@@ -0,0 +1,126 @@
+/**
+ * @file default_init.hpp
+ * @author Chintan Soni
+ * @author Suryoday Basak
+ *
+ * The default initialization policy used by the PSO optimizer.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_PSO_INIT_POLICIES_CONSTRAINED_INIT_HPP
+#define ENSMALLEN_PSO_INIT_POLICIES_CONSTRAINED_INIT_HPP
+
+namespace ens {
+
+/**
+ * The default initialization policy used by the PSO optimizer. It initializes
+ * particle positions uniformly in [-1, 1], the velocities in [0, 1] personal
+ * bests of the particles to the initial positions, and all fitness values to
+ * std::numeric_limits<double>::max().
+ */
+class ConstrainedInit
+{
+ public:
+  /**
+   * Constructor for the DefaultInit policy. The policy initializes particle
+   * posiitons in the range [lowerBound, upperBound]. Defaults to [-1, 1].
+   *
+   * @param lowerBound Lower bound of the position initialization range.
+   * @param upperBound Upper bound of the position initialization range.
+   */
+  ConstrainedInit(const double lowerBound = -1.0,
+              const double upperBound = 1.0):
+      lowerBound(lowerBound),
+      upperBound(upperBound)
+  {
+    /* Nothing to do */
+  }
+
+  /**
+   * The InitializeParticles method of the init policy. Any class that is used
+   * in place of this default must implement this method which is used by the
+   * optimizer.
+   *
+   * @param iterate Coordinates of the initial point for training.
+   * @param numParticles The number of particles in the swarm.
+   * @param particlePositions Current positions of particles.
+   * @param particleVelocities Current velocities of particles.
+   * @param particleFitnesses Current fitness values of particles.
+   * @param particleBestPositions Best positions attained by each particle.
+   * @param particleBestFitnesses Best fitness values attained by each particle.
+   */
+  template<typename FunctionType>
+  void InitializeParticles(FunctionType& function,
```

Comment:
Do you mind to change the name to ``Initialize``, this would be shorter and since this is already in the pso folder the relation between particle is already there.

Reply:
Okay, I'll do that.